google-cloud-spanner/lib/google/cloud/spanner/client.rb

```diff
@@ -1287,6 +1287,20 @@ def create_new_session
           Session.from_grpc grpc, @project.service
         end
 
+        ##
+        # @private
+        # Creates a batch of new session objects of size `session_count`.
+        def batch_create_new_sessions session_count
+          ensure_service!
+          resp = @project.service.batch_create_sessions \
+            Admin::Database::V1::DatabaseAdminClient.database_path(
+              project_id, instance_id, database_id
+            ),
+            session_count,
+            labels: @session_labels
+          resp.session.map { |grpc| Session.from_grpc grpc, @project.service }
```

Comment:
also, these sessions should be shuffled in the session pool, because they sessions created from a single BatchCreateSession rpc will all have the same channel stickiness. adding them in the same order of the pool will cause them to use the same channel for all reads / writes request using the session. 
this will cause a few idle channels, and one or two busy channels.

Reply:
Ruby is single channel so this is likely unnecessary?